app/uk/gov/hmrc/tariffclassificationfrontend/controllers/AttachmentsController.scala

```diff
@@ -110,7 +110,7 @@ class AttachmentsController @Inject()(verify: RequestActions,
   }
 
   def uploadAttachment(reference: String): Action[Either[MaxSizeExceeded, MultipartFormData[TemporaryFile]]] =
-    (verify.authenticated andThen verify.mustHave(Permission.ADD_ATTACHMENT))
+    (verify.authenticated andThen verify.casePermissions(reference) andThen verify.mustHave(Permission.ADD_ATTACHMENT))
```

Comment:
Why did you add verify.casePermissions?   Adding adding attachment is not dependant on case ownership

Reply:
ADD_ATTACHMENT is a `CasePermission`  and requires `verify.casePermissions`

Reply:
I could make  it a  `GlobalPermission` but that seems wrong

Reply:
But it is a global permission - i.e. not dependent on ownership

Reply:
`CasePermission`s  are specific to restricting things on a single `case`. I'd argue Adding attachments is specific to a single case